pennylane/spin/spin_hamiltonian.py

```diff
@@ -689,3 +689,67 @@ def kitaev(n_cells, coupling=None, boundary_condition=False):
         hamiltonian += coeff * (opmap[op1](v1) @ opmap[op2](v2))
 
     return hamiltonian.simplify()
+
+
+def spin_hamiltonian(lattice):
+    r"""Generates a spin Hamiltonian for a custom lattice.
+
+    Args:
+        lattice (Lattice): custom lattice defined with custom_edges
+
+    Raises:
+        ValueError: if ``custom_edges`` are not defined or not defined with operators
```

Comment:
custom_edges is not an input for this function. Maybe you can just say "if the input lattice has no custom edges defined for it." or something similar.